internal/command/image/update.go

```diff
@@ -49,12 +57,36 @@ The update will perform a rolling restart against each VM, which may result in a
 			Name:        "detach",
 			Description: "Return immediately instead of monitoring update progress",
 		},
+		flag.Bool{
+			Name:        "auto-confirm",
+			Description: "Will automatically confirm changes without an interactive prompt.",
+		},
 	)
 
 	return cmd
 }
 
-func runUpdate(ctx context.Context) error {
+func runUpdate(ctx context.Context) (err error) {
+	var (
+		appName = app.NameFromContext(ctx)
+		client  = client.FromContext(ctx).API()
+	)
+
+	app, err := client.GetAppCompact(ctx, appName)
+	if err != nil {
+		return fmt.Errorf("get app: %w", err)
+	}
+
+	switch app.PlatformVersion {
+	case "nomad":
+		return updateImageForNomad(ctx)
+	case "machines":
+		return updateImageForMachines(ctx)
```

Comment:
What do you think about just calling the postgres specific update logic here in the event the app has role `postgres_cluster` and work to keep the image update logic non-specific.

Reply:
Right, we should do our best to keep PG specific orchestration out of the general update logic.